utreexo-p2p-bip.md

```diff
@@ -1,236 +1,360 @@
+This document **does not** describe how to validate blocks and transactions using the provided data, see [Utreexo - Validation Layer](./utreexo-validation-bip.md) for more details.
 
-# Table of Contents
-1. [License](#license)
-2. [Specification](#specification)
-    1. [Block proofs](#block-proofs)
-        1. [Compact leaf data](#compact-leaf-data)
-        2. [Reconstructable Script](#reconstructable-script)
-        3. [Udata](#udata)
-        4. [`block` message](#block-message)
+# Motivation
+
+Utreexo nodes require the inclusion proof to fully validate blocks and transactions.
+Each block has an corresponding inclusion proof with it and this inclusion proof for blocks up to height 906,937 requires an additional 631.85GB, which is roughly 40GB less than the size of the block data.
+Each transaction also has an corresponding inclusion proof with it and for normal transaction relay, the proof is roughly 3 times the size of the transaction.
+It's still reasonable for a single node to download this extra data but little caching goes a long way in reducing the amount of data that one has to download.
+We define the new P2P messages for the inclusion proofs to support caching to reduce bandwidth while also allowing a high bandwidth, low-latency usage.
 
 # License
-This BIP is licensed under the BSD 3-clause license.
 
-# Requirements and Compatibility
+This BIP is licensed under the BSD-3-Clause license.
 
-Nodes implementing Utreexo can choose which messages to support.  There are a number of configurations possible, and this BIP does not restrict nodes to any subsets of messages.  That said, there are two likely types of nodes: Compact State Nodes (CSNs) which have the goal of minimizing data storage and download while performing block validation, and archive and bridge nodes which store more data and provide this data to CSNs.  Bridge nodes, nodes which can add inclusion proofs to mempool transactions, support the same set of messages as CSNs, and in fact should be indistinguishable from CSNs on the network.  Archive nodes can send messages such as block summaries and block proofs.
+# Overview
+
+## Requirements and Compatibility
+
+Nodes implementing Utreexo can choose which messages to support.
+There are a number of configurations possible, and this BIP does not restrict nodes to any subsets of messages.
+That said, there are three likely types of nodes: Compact State Nodes (CSNs) which have the goal of minimizing data storage and download while performing block validation, and archive and bridge nodes which store more data and provide this data to CSNs.
+Bridge nodes are nodes that can add inclusion proofs to mempool transactions, support the same set of messages as CSNs, and are in fact should be indistinguishable from CSNs on the network.
+Archive nodes are able to serve the blocks and the inclusion proofs. However, they are not able to generate the inclusion proofs as they do not keep the full UTXO set.
```

Comment:
Are those meant to be separated paragraphs? Cause they don't have an empty line between then. They are rendered as one big paragraph.

Reply:
No just one big paragraph. I was trying out this thing where every sentence takes up a single line.

Someone said they do this at their company and it works well since you don't need to worry about editing sentences for wrapping.